tp1/tp1.go

```diff
@@ -24,17 +41,66 @@ type Carrito struct {
 // para cada super mercado, cuanto saldria comprar esos productos ahi.
 // Retorna un slice de carritos, donde se tiene uno para cada super mercado.
 func (p Productos) CalcularPrecios(ids ...int) []Carrito {
-	return nil
+	mapCarrito := make(map[string]int)
+	var sliceCarrito []Carrito
+
+	for _, prod := range p {
+		for _, buscado := range ids {
+			if aux, _ := strconv.Atoi(prod[1]); aux == buscado {
```

Comment:
Acá no se está manejando el error, con lo cual si `Atoi` devuelve un error, aux va a valer 0 por default y si hay un buscado que tenga ID 0, entonces va a ser un cálculo incorrecto. Lo adecuado sería chequear el error e ignorar el artículo. 
```suggestion
			if aux, err := strconv.Atoi(prod[1]); aux == buscado && err != nil {
```

Reply:
Hola Marcos!

Hice este cambio, pero cuando lo pruebo falla. Me fije que el if, la parte de **err != nil** me está dando false.

Hice estos prints:
```
	aux, err := strconv.Atoi(productoArray[IDPRODUCTO])
	println("aux:", (aux))
	println("err:", (err))
	println("err != nil:", err != nil)
```

Y me devolvió esto:
aux: 47
err: (0x0,0x0)
err != nil: false

¿Tenes idea que puede ser lo que está pasando ? Porque la conversión a entero la está haciendo bien, pero me devuelve el error de una forma rara

Reply:
Eso esta bien, yo puse mal la condición, tiene que ser err == nil :)

Reply:
Ah joya! Gracias